cs/src/Connections/IRelayClient.cs

```diff
@@ -39,17 +63,25 @@ internal interface IRelayClient
 
     /// <summary>
     /// Configures tunnel SSH session with the given stream.
+    /// If this method succeeds, the SSH session must be connected and ready.
+    /// If this method fails, depending on <see cref="TunnelConnectionOptions.EnableRetry"/> and failure, tunnel client my try reconnecting.
     /// </summary>
     Task ConfigureSessionAsync(Stream stream, bool isReconnect, CancellationToken cancellation);
 
     /// <summary>
-    /// Closes tunnel SSH session due to an error or exception.
+    /// Closes tunnel SSH session due to an error or exception happened during connection.
+    /// Depending on <see cref="TunnelConnectionOptions.EnableRetry"/> and nature of the error,
+    /// the connection may or may not do another attempt.
     /// </summary>
     Task CloseSessionAsync(SshDisconnectReason disconnectReason, Exception? exception);
 
     /// <summary>
     /// Refresh tunnel access token. This may be useful when the Relay service responds with 401 Unauthorized.
     /// </summary>
+    /// <exception cref="ObjectDisposedException">If the connection is disposed or disconnected.</exception>
+    /// <exception cref="ArgumentException">If current status is not <see cref="ConnectionStatus.Connecting"/>.</exception>
+    /// <exception cref="UnauthorizedAccessException">If the refreshed token is expired.</exception>
+    /// <exception cref="OperationCanceledException">If refresh is cancelled or connection disposed.</exception>
```

Comment:
If connection is disposed how is it determined which of these it will be between ObjectDisposedException and OperationCanceledException

Reply:
It depends on the timing. When relay client is disposed, it cancels its connection operation. If that cancellation is observed first, the method may throw `OperationCancelledException`. If disposal happens a little bit later, it may become `ObjectDisposedException`. Anyway, this is used by `RelayTunnelConnector` where it's then always converted to `ObjectDisposedException` if the relay client is disposed, `OperationCancelledException` if the consumer cancelled connection, or `UnauthorizedAccessException` in other cases.